lesson17/main.js

```diff
@@ -0,0 +1,123 @@
+const jsonUrl = "https://jsondata.okiba.me/v1/json/YbIZe211030061143";
+const slideshowWrap = document.getElementById("js-slideshow");
+const ul = document.getElementById("js-img-list");
+
+const createElementWithClassName = (element, name) => {
+  const createdElement = document.createElement(element);
+  createdElement.classList.add(name);
+  return createdElement;
+};
+
+const createLoader = () => {
+  const loader = document.createElement("div");
+  const loaderImage = document.createElement("img");
+  loader.id = "js-loader";
+  loaderImage.src = "img/loading-circle.gif";
+  slideshowWrap.appendChild(loader).appendChild(loaderImage);
+};
+
+const loading = () => {
+  const loader = document.getElementById("js-loader");
+  loader.classList.add("loading");
+};
```

Comment:
Just a question,
I felt that separating "createLoader" and "loading" made it more readable. On the other hand, it felt easier to put them together to reduce the amount of code. I would like to know if there are any other reasons for separating these.

Reply:
Thank you for your question :)

I used to implement them together.
But if I run this `loading` in other situation when I need loading, every time I made copy loading which has same id, so I can't control each loading if I want to.
And also showing/hiding lodaing can be done by just manipulating the class, but if you implement them together, it will create elements etc every time, which is not good for performance.

Reply:
I understand very well. Thank you for always helping me to learn.